File: Data Structures & Algorithms/jump-game-ii/submission-4.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int i=0,j=nums.size();
        int steps=0;
if(j==1){
    return 0;
}
        while(i<j){
            if(nums[i]==0)return 0;
            if(nums[0]>=j-1)return 1;
           steps++;
            int a=i;
            int n=nums[a];
            int m=-1;
            for(int x=a;x<=a+n; x++){
                if(nums[x]+x>=j-1)return steps+1;
                if(nums[x+nums[x]]==0)continue;
                if(nums[x]>m){
                    m=nums[x];
                    i=x+nums[x];
                }
                // m=max(nums[x],m);
            }
            
        }
        return steps;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums.sort()
        n=len(nums)
        if n==0:
            return 0
        i=1
        ans=1
        x=1
        while i<n:
            if nums[i-1]==nums[i]:
                i+=1
                continue
            if nums[i-1]+1==nums[i]:
                ans+=1
                print(ans)
            else:
                ans=1
            x=max(ans,x)
            i+=1
        return x 